src/sparselm/model/_base.py

```diff
@@ -151,20 +240,39 @@ def _gen_objective(self, X: ArrayLike, y: ArrayLike):
                 Covariate/Feature matrix
             y (ArrayLike):
                 Target vector
+            beta (cp.Variable):
+                cp.Variable representing the estimated coefs_
+            parameters (SimpleNamespace): optional
+                SimpleNamespace with cp.Parameter objects
+            auxiliaries (SimpleNamespace): optional
+                SimpleNamespace with auxiliary cvxpy objects
 
         Returns:
             cvpx Expression
         """
-        return None
+        return
 
-    def _gen_constraints(self, X: ArrayLike, y: ArrayLike):
+    def _generate_constraints(
```

Comment:
Maybe call this "generate canonicals?"

Reply:
These are only constraints, "canonicals" are basically all cvxpy objects, including problem, objective, parameters and auxiliary expressions.